roachprod: clean up use of SyncedCluster

RaduBerinde opened the pull request:
#### roachprod: move quiet determination out of the library

Moving the logic of automatically enabling Quiet in non-terminal
output.

Release note: None

#### roachprod: clean up use of SyncedCluster

`SyncedCluster` is currently used to pass the cluster name (with
optional node selector) and the settings. This is a misuse of the type
and complicates things conceptually.

This change separates out the relevant settings into a new struct
`ClusterSettings`. All commands now pass the cluster name and the
`ClusterSettings` instead of passing a `SyncedCluster`.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/72442)
<!-- Reviewable:end -->

rail reviewed (changes requested):
> Reviewed 2 of 2 files at r1, 7 of 7 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/72442)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @healthy-pod, @RaduBerinde, and @stevendanna)

___
*[pkg/roachprod/install/nodes.go, line 75 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/72442#-Mo-_Q_D324j8DNFDXFM:-Mo-_Q_D324j8DNFDXFN:b-j12ztf) ([raw file](https://github.com/cockroachdb/cockroach/blob/66ee5d7ef6523fdc23ba2a28b7ef18773af14577/pkg/roachprod/install/nodes.go#L75)):*
> ```Go
> 	for _, n := range nodes {
> 		if n < 1 {
> 			return nil, fmt.Errorf("invalid node selector '%s', node values start at 1")
> ```

Missing `n` as a parameter to `fmt.Errorf`. Otherwise LGTM.


<!-- Sent from Reviewable.io -->

RaduBerinde committed:
roachprod: move quiet determination out of the library

Moving the logic of automatically enabling Quiet in non-terminal
output.

Release note: None

RaduBerinde committed:
roachprod: clean up use of SyncedCluster

`SyncedCluster` is currently used to pass the cluster name (with
optional node selector) and the settings. This is a misuse of the type
and complicates things conceptually.

This change separates out the relevant settings into a new struct
`ClusterSettings`. All commands now pass the cluster name and the
`ClusterSettings` instead of passing a `SyncedCluster`.

Release note: None

RaduBerinde reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/72442)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @healthy-pod, @rail, and @stevendanna)

___
*[pkg/roachprod/install/nodes.go, line 75 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/72442#-Mo-_Q_D324j8DNFDXFM:-Mo04uA-DUh7B6nC_oXC:b-qqx0w8) ([raw file](https://github.com/cockroachdb/cockroach/blob/66ee5d7ef6523fdc23ba2a28b7ef18773af14577/pkg/roachprod/install/nodes.go#L75)):*
<details><summary><i>Previously, rail (Rail Aliiev) wrote…</i></summary><blockquote>

Missing `n` as a parameter to `fmt.Errorf`. Otherwise LGTM.
</blockquote></details>

Done. TFTR!


<!-- Sent from Reviewable.io -->

rail reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 7 of 7 files at r3, 7 of 7 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/72442#-:-Mo05gpR-_zZ7dbCMAnt:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @healthy-pod and @stevendanna)



<!-- Sent from Reviewable.io -->

RaduBerinde commented:
bors r+
